### ISSUE TITLE
add frontend

### ISSUE THREAD

**Issue by tldev-de** [tldev-de/dockinv#4]
There should be a frontend with the following features:

**Hosts**
Table with:
- name
- last successful agent call
- amount of docker container
- amount of docker images
- amount of eol images
- aggregated trivy findings

**Container**
from the hosts view there should be a possibility to see the containers of one host. this could be a table with:
- container name
- image name / tag
- status
- result of xeol
- result of trivy

**Images**
It would be nice to have an overview of all images used in any host. this could be a table with:
- image name
- result of xeol
- result of trivy
- amount of usage by containers

**Comment by JulianSobott** [comment]
Should it also be possible to add new hosts via the frontend? On this page, we could then add documentation on how to install an agent.

**Comment by JulianSobott** [comment]
What technology should we use?

- Python+Jinja2
  - Fast prototyping
  - Easy to use
  - Whole project requires only python skills
- React
  - Most popular frontend framework
  - Some features may be easier to implement
    - Live view
    - Table filters (by host, tag, ...)
    - Searches
 
I think we should start with python+jinja2 and keep it very simple. If this works, and we need more features, we can switch to something else.

**Comment by tldev-de** [comment]
> Should it also be possible to add new hosts via the frontend? On this page, we could then add documentation on how to install an agent.

yes, that would be awesome! at the moment hosts can be managed via flask api. this is a way, that is (from my perspective) a good basic start and worth preserving in future developing. a second way in frontend to manage hosts would be awesome!


> I think we should start with python+jinja2 and keep it very simple. If this works, and we need more features, we can switch to something else.

I agree that python + jinja is a good simple start! I think we can achieve all goals with a server side rendered backend. IN later development an API might be useful, this could be a good starting point of a react based frontend.

**Comment by JulianSobott** [comment]
Implemented some basic frontend. Next steps would be to test it and refine the requirements.
For example, showing trivy or xeol findings more detailed would be nice, but these files are huge and we must find out which information is actually relevant.

![Image](https://github.com/user-attachments/assets/5e374930-0f26-4362-a59d-d3beaf61208f)
![Image](https://github.com/user-attachments/assets/f4f13e78-9236-44b9-bd01-1c9b0e9b8c7b)
![Image](https://github.com/user-attachments/assets/942698f9-3bce-46d5-afa8-a886c3ca36da)
